### PR TITLE
chore(examples): fix compilation errors and use maven on sdk examples

### PR DESCRIPTION
**Pull request by kevinkowa**
This PR fixes the examples and changes from using gradle to maven. It also updates the readme to show how to install the dependencies.

Giving credit to @ima1mai on the work in the discovery example 🎉  (https://github.com/watson-developer-cloud/java-sdk/pull/1193)

### CODE REVIEW

File: examples/src/main/java/com/ibm/watson/speech_to_text/v1/MicrophoneWithWebSocketsExample.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import com.ibm.cloud.sdk.core.security.Authenticator;
 import com.ibm.cloud.sdk.core.security.IamAuthenticator;
 import com.ibm.watson.speech_to_text.v1.model.RecognizeOptions;
```
Comment (Hsaylor) [inline examples/src/main/java/com/ibm/watson/speech_to_text/v1/MicrophoneWithWebSocketsExample.java:18]: I don't think you need this import anymore

File: examples/src/main/java/com/ibm/watson/speech_to_text/v1/RecognizeUsingWebSocketsExample.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import com.ibm.cloud.sdk.core.security.Authenticator;
 import com.ibm.cloud.sdk.core.security.IamAuthenticator;
 import com.ibm.watson.speech_to_text.v1.model.RecognizeOptions;
```
Comment (Hsaylor) [inline examples/src/main/java/com/ibm/watson/speech_to_text/v1/RecognizeUsingWebSocketsExample.java:18]: same thing here

File: examples/src/main/java/com/ibm/watson/speech_to_text/v1/MicrophoneWithWebSocketsExample.java (hunk 1 above)
Comment (kevinkowa) [reply examples/src/main/java/com/ibm/watson/speech_to_text/v1/MicrophoneWithWebSocketsExample.java:18]: good catch!

**Review by Hsaylor — DISMISSED** [review]
lgtm

File: examples/src/main/java/com/ibm/watson/assistant/v1/tone_analyzer_integration/ToneDetection.java [hunk 3]
```diff
@@ -58,14 +59,14 @@ private ToneDetection() {}
    *     toneAnalyzerPayload
    */
   public static Map<String, Object> updateUserTone(
-      Context context, ToneAnalysis toneAnalyzerPayload, boolean maintainHistory) {
+          Context context, ToneAnalysis toneAnalyzerPayload, boolean maintainHistory) {
```
Comment (apaparazzi0329) [inline examples/src/main/java/com/ibm/watson/assistant/v1/tone_analyzer_integration/ToneDetection.java:62]: Any reason for the indentation here?

**Review by apaparazzi0329 — APPROVED** [review]
LGTM although while you are at it I would also remove C&C and PI examples since those services are officially shutdown as well

**Review by apaparazzi0329 — APPROVED** [review]
LGTM

**Pull request merged**